src/common/utxobased/engine/makeUtxoEngineState.ts

```diff
@@ -184,44 +199,96 @@ interface CommonArgs {
   walletInfo: EdgeWalletInfo
   walletTools: UTXOPluginWalletTools
   processor: Processor
-  blockBook: BlockBook
   emitter: EngineEmitter
-  addressesToWatch: Set<string>
+  addressesToWatch: Map<string, ShortPath>
+  taskCache: TaskCache
   onAddressChecked: () => void
   mutexor: Mutexor
 }
 
-interface OnNewBlockArgs extends CommonArgs {}
+const run = async (
+  args: CommonArgs,
+  executor: (args: FormatArgs) => unknown
+): Promise<void> => {
+  const formatsToProcess = getWalletSupportedFormats(args.walletInfo)
+  for (const format of formatsToProcess) {
```

Comment:
Replace with format branch from below.

Reply:
Same as format branch above, this is changed after rebasing on current master, so has to be fixed independently upon rebase.